### ISSUE TITLE
False positive Mypy can't infer an empty dicts when using in `dict.get(fallback=...)`

### ISSUE THREAD

**Issue by pavelzw** [python/mypy#14728]
**Bug report**

When passing an empty dict to a function, in this example `dict.get`, it cannot be inferred correctly. 

**To Reproduce**

```python
from typing import Any

# this doesn't work
d: dict[str, Any] = {'a': {'b': 42}}.get('a', {})

# this works
empty: dict[str, Any] = {}
d2: dict[str, Any] = {'a': {'b': 42}}.get('a', empty)
```

[Mypy playground](https://mypy-play.net/?mypy=latest&python=3.11&gist=e773ae2f458c7e1ff14c4083c3bad078)

**Expected Behavior**

Mypy shouldn't throw errors.

**Actual Behavior**

```
main.py:4: error: Argument 2 to "get" of "dict" has incompatible type "Dict[<nothing>, <nothing>]"; expected "Union[Dict[str, int], Dict[str, Any]]"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.0.0
- Python version used: 3.11

**Comment by pavelzw** [comment]
Fixed in mypy 1.6.0
